### PR TITLE
TGUI Search component and disposal / mail chute UI

### PR DESCRIPTION
**Pull request by Sovexe**
## About the PR 
- Adds Search.js by Mordent
- Adds TGUI disposal and mail chute UIs by me (with lots of help from Mordent setting up mail dests and general code prettiness)

edit: I've redesigned my initial submission significantly to a newer cleaner feel

image

Images below are of previous revision
NOTE: the Disposal Handle label in the images has been renamed Chute Handle, very small change so i didnt make new pics
https://user-images.githubusercontent.com/33204415/90360958-062c3f00-e02b-11ea-9901-c078c8f66c4c.png
https://user-images.githubusercontent.com/33204415/90360991-16441e80-e02b-11ea-848c-13fbcac33d9f.png
https://user-images.githubusercontent.com/33204415/90361022-2825c180-e02b-11ea-8ef4-87a005e21dc6.png
https://user-images.githubusercontent.com/33204415/90361079-4f7c8e80-e02b-11ea-87ee-be95da7f59ce.png
https://user-images.githubusercontent.com/33204415/90361144-75099800-e02b-11ea-991e-48f8702b55b8.png

There is a known issue with text input fields that should be fixed before this is merged. While it is functional as is, the user experience would be poor when searching for mail destinations.

## Why's this needed? 
- Search is a new component by Mordent
- More TGUI

## Changelog

```
(u)Sovexe:
(+)Converted disposal and mailchute to TGUI
```

### CODE REVIEW

File: tgui/packages/tgui/components/Search.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    placeholder = 'Search',
+    selectedOption = null,
+  } = props;
+  const handleSearch = (e, value) => {
+    onSearch(value);
+  };
+  return (
+    <div>
+      <Input
+        autoFocus
```
[the remark this answers is not in the document]
Comment (Sovexe) [reply tgui/packages/tgui/components/Search.js]: done

File: code/WorkInProgress/recycling/disposal_chute.dm [hunk 2]
```diff
@@ -528,7 +466,7 @@
 	suicide(var/mob/living/carbon/human/user as mob)
 		if (!istype(user) || !src.user_can_suicide(user))
 			return 0
-		if (src.mode != 2)//!hasvar(user,"organHolder")) I will END YOU
+		if (src.mode != DISPOSAL_CHUTE_CHARGED)//!hasvar(user,"organHolder")) I will END YOU
```
[the remark this answers is not in the document]
Comment (Sovexe) [reply code/WorkInProgress/recycling/disposal_chute.dm]: removed

File: code/WorkInProgress/recycling/disposal_chute.dm [hunk 3]
```diff
@@ -360,7 +300,7 @@
 		if (status & BROKEN)
 			icon_state = "disposal-broken"
 			ClearAllOverlays()
-			mode = 0
+			mode = DISPOSAL_CHUTE_OFF
```
[the remark this answers is not in the document]
Comment (Sovexe) [reply code/WorkInProgress/recycling/disposal_chute.dm:303]: I believe it's fine as this check
```
	process()
		if(status & BROKEN)			// nothing can happen if broken
			return
```
prevents it from ever calling the `process` parent and `use_power`

File: tgui/packages/tgui/components/Search.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+ * @license ISC
+ */
+
+import { classes } from 'common/react';
+import { Input } from './Input';
+import { Section } from './Section';
+
+export const Search = props => {
+  const {
+    currentSearch,
```
[the remark this answers is not in the document]
Comment (Sovexe) [reply tgui/packages/tgui/components/Search.js]: i added className to props

unsure if I needed to do anything else with it so I'm not resolving this comment

**Comment by ZeWaka** [comment]
probly want the text box thing fixed b4 merge

**Comment by Sovexe** [comment]
I've redone the design on this to have a look and feel closer to that of the other UIs that I have designed

Because the code was changed so significantly in the process I am re-requesting a review from Mordent

**Review by mordent-goonstation — APPROVED** [review]

**Pull request merged**